### PR TITLE
fix: Update js-langchain template

### PR DESCRIPTION
**Pull request by jirispilka**
Update js-langchain template:
- Update dependencies.
- Move hnswlib and ApifyDatasetLoader to `langchain/community` and refactor code accordingly.
- Remove deprecated `RetrievalQaChain`.
- Replace `console.log()` with `Apify.log()`.
- Add eslint.

### CODE REVIEW

**Comment by jirispilka** [comment]
I tested it at my local. I don't see the warning in the CI, except
```
 ›   Warning: Could not find typescript. Please ensure that typescript is a 
 ›   devDependency. Falling back to compiled source.
```
But I don't want to add typescript into the dependencies. 

image

File: templates/js-langchain/README.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ## LangChain.js template
 
 > LangChain is a framework for developing applications powered by language models.
-> 
+>
```
Comment (B4nan) [inline templates/js-langchain/README.md:4]:
```suggestion
```

File: templates/js-langchain/README.md [hunk 2]
```diff
@@ -33,7 +33,7 @@ To be able to run this template both locally and on the Apify platform, you need
 ## Production use
 
 > This serves purely as an example of the whole pipeline.
-> 
+>
```
Comment (B4nan) [inline templates/js-langchain/README.md:36]:
```suggestion
```

**Review by B4nan — APPROVED** [review]
nice, just one tiny nit

**Comment by jirispilka** [comment]
> nice, just one tiny nit

oh, thanks!

**Pull request merged**